### PR TITLE
ci(monorepo): run app tests when app-shell-odd changes

### PR DESCRIPTION
**Pull request by shlokamin**
# Overview

Oops my bad I noticed app tests arent running when there are code changes in `app-shell-odd`

# Changelog

- Run app checks when app-shell-odd changes

# Risk assessment

Low

### CODE REVIEW

**Review by sfoster1 — APPROVED** [review]
Cool but like... should they be? This will increase our runner load by running the full app test stack again on all three OS's for every ODD shell change. Can we make a new github action that runs only on linux and is triggered by app ui and dependency changes, and odd-shell changes? that way we separate them, and if we make our tests work differently (or add end-to-end tests) for the ODD, we have a clear place to drop them.

**Comment by shlokamin** [comment]
@sfoster1 good call — ill make a separate workflow

**Review by koji — APPROVED** [review]
lgtm

**Comment by shlokamin** [comment]
i think we might use even more CI resources by creating a new workflow. if we create a new workflow, we will do the following anytime the app, app shell odd, components, shared data, or discovery client are changed:

- install dependencies (`make setup-js`)
- run unit tests for `app-shell-odd`
- build the app for the ODD


let's consider the two situations when someone modifies code in the app directory (which will be way more common than changing code in app-shell-odd):

1. we create a new workflow for app-shell-odd

the new workflow would run in addition to the current full stack app test stack. That is one extra make setup (expensive), and then testing the app shell odd (cheap), and then building a linux arm 64 dist (expensive)

2. we do what I have currently

we do the setup which would already be happening anyways, we run tests for the app-shell-odd (cheap), we build a linux arm 64 dist (expensive)


so in the more common scenario of changing app, component, or shared data code, we spend less resources.

File: .github/workflows/app-test-build-deploy.yaml [hunk 1]
```diff
@@ -165,6 +165,13 @@ jobs:
           name: 'opentrons-${{ matrix.os }}'
           path: app-shell/dist/publish
 
+      # build the ODD app
+      - if: ${{ github.event_name != 'pull_request' && matrix.os == 'ubuntu-22.04' }}
```
Comment (shlokamin) [inline .github/workflows/app-test-build-deploy.yaml:169]: this will only get run once by the ubuntu runner

**Review by sfoster1 — APPROVED** [review]
Okay looks good!

**Pull request merged**